execution/gethexec/timeboost_sequencer.go

```diff
@@ -147,40 +180,53 @@ func TimeboostSequencerConfigAddOptions(prefix string, f *flag.FlagSet) {
 	TimeboostBridgeConfigAddOptions(prefix+".timeboost-bridge-config", f)
 }
 
-func NewTimeboostSequencer(execEngine *ExecutionEngine, l1Reader *headerreader.HeaderReader, channel chan DelayedMessageCommand, configFetcher TimeboostSequencerConfigFetcher) (*TimeboostSequencer, error) {
+func NewTimeboostSequencer(execEngine *ExecutionEngine, l1Reader *headerreader.HeaderReader, delayedSequencer decentralizedtimeboost.TimeboostDelayedSequencerInterface, configFetcher TimeboostSequencerConfigFetcher, timeboostBridge *TimeboostBridge) (*TimeboostSequencer, error) {
 	return &TimeboostSequencer{
-		config:     configFetcher,
-		execEngine: execEngine,
-		l1Reader:   l1Reader,
-		nonceCache: newNonceCache(configFetcher().NonceCacheSize),
-		timeboostTxnListener: &TimeboostBridge{
-			config:     configFetcher().TimeboostBridgeConfig,
-			grpcClient: nil,
-		},
+		config:              configFetcher,
+		execEngine:          execEngine,
+		l1Reader:            l1Reader,
+		nonceCache:          newNonceCache(configFetcher().NonceCacheSize),
+		timeboostBridge:     timeboostBridge,
 		delayedMessagesRead: 0,
-		channel:             channel,
+		delayedSequencer:    delayedSequencer,
 	}, nil
 }
 
-func (s *TimeboostSequencer) handleDelayedMessages(delayedMsgsRead uint64) bool {
-	log.Info("sending delayed messages", "read", delayedMsgsRead)
-	for {
-		s.channel <- DelayedMessageCommand{delayedMsgsRead}
-		delayedMsgNum, err := s.execEngine.NextDelayedMessageNumber()
-		log.Info("next delayed msg num", "num", delayedMsgNum)
+func (s *TimeboostSequencer) sequenceDelayedMessages(ctx context.Context, tx *timeboostTransactionQueueItem) error {
```

Comment:
should move this function to a delayed sequencer file or something to not make this file too big